components/vault/DefaultVaultHeaderControl.tsx

```diff
@@ -1,40 +1,35 @@
-import BigNumber from 'bignumber.js'
-import { useAppContext } from 'components/AppContextProvider'
-import { VaultContainerSpinner, WithLoadingIndicator } from 'helpers/AppSpinner'
-import { useObservableWithError } from 'helpers/observableHook'
+import { IlkDataList } from 'blockchain/ilks'
 import React from 'react'
 
-import { DefaultVaultHeaderLayout, DefaultVaultHeaderLayoutProps } from './DefaultVaultHeaderLayout'
+import { Vault } from '../../blockchain/vaults'
+import { DefaultVaultHeaderLayout } from './DefaultVaultHeaderLayout'
 
 export interface DefaultVaultHeaderControlProps {
-  vaultId: BigNumber
+  vault: Vault
+  ilkDataList: IlkDataList
 }
 
-export function DefaultVaultHeaderControl(props: DefaultVaultHeaderControlProps) {
-  const { vaultId } = props
-  //console.log('Rendering DefaultVaultHeaderControl', vaultId.toString())
-  const { vault$, ilkDataList$ } = useAppContext()
-  const vaultDataWithError = useObservableWithError(vault$(vaultId))
-  const ilksDataWithError = useObservableWithError(ilkDataList$)
-  return (
-    <WithLoadingIndicator
-      variant="DefaultVaultHeaderControl"
-      value={[vaultDataWithError.value, ilksDataWithError.value]}
-      customLoader={<VaultContainerSpinner />}
-    >
-      {([vaultData, ilkDataList]) => {
-        const ilk = ilkDataList.filter((x) => x.ilk === vaultData.ilk)[0]
+export function DefaultVaultHeaderControl({ ilkDataList, vault }: DefaultVaultHeaderControlProps) {
+  const ilk = ilkDataList.filter((x) => x.ilk === vault.ilk)[0]
 
-        const props: DefaultVaultHeaderLayoutProps = {
-          id: vaultId,
-          debtFloor: ilk.debtFloor,
-          liquidationPenalty: ilk.liquidationPenalty,
-          liquidationRatio: ilk.liquidationRatio,
-          stabilityFee: ilk.stabilityFee,
-        }
+  const vaultHeaderProps = {
+    id: vault.id,
+    debtFloor: ilk.debtFloor,
+    liquidationPenalty: ilk.liquidationPenalty,
+    liquidationRatio: ilk.liquidationRatio,
+    stabilityFee: ilk.stabilityFee,
+  }
 
-        return <DefaultVaultHeaderLayout {...props} />
-      }}
-    </WithLoadingIndicator>
-  )
+  const guniHeaderProps = {
+    id: vault.id,
+    stabilityFee: ilk.stabilityFee,
+    debtFloor: ilk.debtFloor,
+  }
+
+  const headerProps =
+    vault.ilk === 'GUNIV3DAIUSDC1-A' || vault.ilk === 'GUNIV3DAIUSDC2-A'
```

Comment:
```suggestion
['GUNIV3DAIUSDC1-A', 'GUNIV3DAIUSDC2-A'].includes(vault.ilk)
```

A more "scalable solution, easy to extend